### PR TITLE
feat(token): add p-token (SIMD-0266) and missing SPL Token instructions

### PR DESCRIPTION
**Pull request by hxuan190**
Add support for 7 missing token instructions to the SPL Token program:

Instructions 21-24 (existing on-chain, missing from solana-go):
- GetAccountDataSize (ID 21)
- InitializeImmutableOwner (ID 22)
- AmountToUiAmount (ID 23)
- UiAmountToAmount (ID 24)

P-token exclusive instructions (SIMD-0266 / Pinocchio):
- WithdrawExcessLamports (ID 38)
- UnwrapLamports (ID 45)
- Batch (ID 255)

Updated instructions.go with new constants, InstructionIDToName entries, extended InstructionImplDef for IDs 0-24, and added pTokenInstructionMap with custom DecodeInstruction dispatch for non-contiguous p-token IDs.

Verified on Solana testnet (SIMD-0266 feature gate activated at slot 396236256, address ptokFjwyJtrwCa9Kgo9xoDS59V4QccBGEaRFnRPnSdP):

Simulation tests (IDs 20-24):
- InitializeMint2: 364 CUs
- GetAccountDataSize: 179 CUs
- AmountToUiAmount: 615 CUs
- UiAmountToAmount: 589 CUs

Send+confirm tests (p-token exclusive):
- WithdrawExcessLamports (ID 38): 3C9iNnUPPaqRCA1RSD5tXDpD7ZW72VQYsKYm1j3voEYcH4UCfzKURan63Df2jdfX3gzPzEHZDu6eFfnTf2K4H98d
- UnwrapLamports partial (ID 45): zPrV9K6wLqnES4RBDAQs15S8MgvC48QjsLZAmPoimvY6vmQkh8yi3XBJYqQBkpo28HpiNdhuPL3N65Dqf1ovhWn
- UnwrapLamports all (ID 45): 5VVDC6PeWo5sAAUhbc3xRRiZDMo1t5Y6c1VDktBqLLhXZtri1HshL9y7TmEu9BBZEpiQCdAmULazfNjTfeHd5ujH
- Batch with 2 transfers (ID 255): 2y1jC5rXbQvvym1bM5ukRcQfiovUQPi4fbcryixXpjTe5fc2UDGvvgurZqLNx7DEYd7NHHwEnEt6oPX4anV9kXnh

### CODE REVIEW

**Comment by hxuan190** [comment]
This feature is live on mainnet & made some test on devnet with test keypair ```Ct1DVKNkhZV5qvXqu5Hr7rN9NfVDWpVQFxaUqHjhfpG2```

File: programs/token/GetAccountDataSize.go
Comment (sonicfromnewyoke) [inline programs/token/GetAccountDataSize.go:1]: this instruction delivesr their result via `sol_get_return_data`

File: programs/token/UnwrapLamports.go
Comment (sonicfromnewyoke) [inline programs/token/UnwrapLamports.go:1]: this instruction delivesr their result via `sol_get_return_data`

File: programs/token/AmountToUiAmount.go
Comment (sonicfromnewyoke) [inline programs/token/AmountToUiAmount.go:1]: this instruction delivesr their result via `sol_get_return_data`

**Review by sonicfromnewyoke — COMMENTED** [review]
Overall looks good, left some comments after addressing them and adding some tests it will be nice to add it

**Comment by hxuan190** [comment]
@sonicfromnewyoke can you tell me more information about your review? confused a little bit

**Comment by sonicfromnewyoke** [comment]
> @sonicfromnewyoke can you tell me more information about your review? confused a little bit

sorry, my inline notes were too terse... i meant the next one:

there is no Go-side helper for the return-data instructions. Marked ixns deliver their result  via `sol_get_return_data`. The struct builders here only handle the request side - so a caller still has to hand-roll the LE-u64 / UTF-8 decoding after sim txn. That was the point of my three inline comments: there we eaither need a  small ParseReturnData helper per instruction (or a single typed dispatcher) so the SDK is symmetric or simply drop those inxns because not so many guys are using them this way.

`Batch.UnmarshalWithDecoder` drops `accountCount`- the whole reason the on-chain format encodes per sub-ix is so a decoder can slice the flat account list back into per-sub-ix accounts.

also, the novel logic has no test, for example round-trip fuzz tests for Batc missing - the file only asserts ID constants

despite comments above, this is a solid PR overall 💪

File: programs/token/Batch.go [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+					instructionBranch.Child("Accounts").ParentFunc(func(accountsBranch ag_treeout.Branches) {
+						for i, acc := range inst.AccountMetaSlice {
+							accountsBranch.Child(ag_format.Meta(fmt.Sprintf("[%v]", i), acc))
+						}
+					})
+				})
+		})
+}
+
+func (obj Batch) MarshalWithEncoder(encoder *ag_binary.Encoder) (err error) {
```
Comment (HealthyBuilder) [inline programs/token/Batch.go:151]: Batch.MarshalWithEncoder silently truncates if a sub-instruction has > 255 accounts or > 255 bytes of data — please return an explicit error.

**Comment by hxuan190** [comment]
> > @sonicfromnewyoke can you tell me more information about your review? confused a little bit
> 
> sorry, my inline notes were too terse... i meant the next one:
> 
> there is no Go-side helper for the return-data instructions. Marked ixns deliver their result via `sol_get_return_data`. The struct builders here only handle the request side - so a caller still has to hand-roll the LE-u64 / UTF-8 decoding after sim txn. That was the point of my three inline comments: there we eaither need a small ParseReturnData helper per instruction (or a single typed dispatcher) so the SDK is symmetric or simply drop those inxns because not so many guys are using them this way.
> 
> `Batch.UnmarshalWithDecoder` drops `accountCount`- the whole reason the on-chain format encodes per sub-ix is so a decoder can slice the flat account list back into per-sub-ix accounts.
> 
> also, the novel logic has no test, for example round-trip fuzz tests for Batc missing - the file only asserts ID constants
> 
> despite comments above, this is a solid PR overall 💪

 Good catch — these instructions don't write to an account, they emit the result via sol_get_return_data. I've added decode helpers so callers can parse the return data after simulating the tx:
 - ParseGetAccountDataSizeResult([]byte) (uint64, error) — little-endian u64
 - ParseAmountToUiAmountResult([]byte) string — UTF-8 string
 - ParseUiAmountToAmountResult([]byte) (uint64, error) — little-endian u64 (added this one too, since it has the same return-data behaviour)

 Each is covered by tests. Verified the wire formats against the p-token processors (set_return_data calls).

File: programs/token/UnwrapLamports.go
Comment (hxuan190) [reply programs/token/UnwrapLamports.go:1]: I think this comment may have been intended for ```UiAmountToAmount``` rather than ```UnwrapLamports``` 
I checked ```pinocchio/program/src/processor/unwrap_lamports.rs``` and it doesn't call ```set_return_data```, whereas ```ui_amount_to_amount.rs``` does. I've added a return-data decoder for ```UiAmountToAmount``` (and ```GetAccountDataSize/AmountToUiAmount```). Let me know if ```UnwrapLamports``` is meant to return data in a version I'm missing.

File: programs/token/Batch.go (hunk 1 above)
Comment (hxuan190) [reply programs/token/Batch.go:151]: **Fixed** 
MarshalWithEncoder now returns an explicit error when a sub-instruction exceeds 255 accounts or 255 data bytes, instead of silently truncating through the uint8 cast. The error names the sub-instruction index and the actual size. Added tests covering the 256-account, 256-byte, and 255-boundary cases.

**Comment by hxuan190** [comment]
hi @HealthyBuilder can you review it
